src/util.ts

```diff
@@ -494,35 +496,81 @@ export function selectDefFrom4byteABI (abiData: any[], selector: string) {
   }
 }
 
-async function fetchSupportedChainData (address: string, supportedChain: number) {
-  const url = buildUrlForSupportedChainAndAddress({ address, supportedChain })
-  return superagent
-    .get(url)
-    .then(async res => {
-      if (res && res.body && res.body.result) {
-        return JSON.parse(res.body.result)
+export async function fetchWithTimeout (url, options) {
+  const { timeout } = options;
+  const controller = new AbortController();
+  const timeoutId = setTimeout(() => controller.abort(), timeout);
+  const response = await fetch(url, {
+    ...options,
+    signal: controller.signal,
+  });
+  clearTimeout(timeoutId);
+  return response;
+}
+
+async function fetchAndCache (url, opts?) {
+  try {
+    if (globalThis.caches && globalThis.Request) {
+      const cache = await caches.open('gp-calldata');
+      const request = new Request(url, opts);
+      const match = await cache.match(request);
+      if (match) {
+        return match;
```

Comment:
One thing I just thought of is that 4byte responses may change over time. Someone may add a new (colliding) function definition and that should require a new cache. 

There are two things we can do to ameliorate this. They are sort of but not exactly mutually exclusive:

1. If `match` is >1 month old, force a new request (not sure if you have insight into age of the cached item)
2. We could just not cache 4byte results

Thoughts?

Reply:
Btw to clarify: we don't have to worry about Etherscan et al responses changing because those are tied to a specific contract deployed at a specific address. By contrast, 4byte responses are just function definitions, i.e. are not anchored in blockchain state.

Reply:
This API doesn't offer time-based expiration or the ability to save date metadata. So I guess we just can't cache those requests. 

How can we rely on that data if that's the case?

Reply:
[6575589](https://github.com/GridPlus/gridplus-sdk/pull/469/commits/6575589b71f6f6b8f63c273c13d3a8d9e523681c)